LSPS5/README.md

```diff
@@ -0,0 +1,264 @@
+MIME type `application/json`, containing a JSON object like
+the below:
+
+```JSON
+{
+  "version": 1,
+  "lsp": "023456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef012",
+  "details": {
+    "reason": 1,
+    "pushed_at": "2023-04-25T14:27:03.842Z",
+    "zbase": "blahblah"
+  }
+}
+```
+
+`version` is a required integral number, the version of the
+API that was specified in the `lsps5.setwebhook` call.
+
+`lsp` is a required node ID, the node ID of the LSP that is
+making this push request.
+
+`details` is a required object whose format is dependent on
+`version`.
+
+* For `version` 1, `details` has the fields:
+  * `reason` is a required integral number, which is one
+    of the reason codes defined below.
+  * `pushed_at` is a required datetime, which is the time
+    at the LSP when it generated this push request.
+  * `zbase` is a required signature from the LSP, signing
+    the message defined below.
+
+For `version` 1 `zbase` signatures, the signature signs
+the following JSON string:
+
+    "LSPS5:Version 1: Reason ${reason} at time ${pushed_at}."
+
+Where `${reason}` is the shortest decimal text form of the
+`reason` code, and `${pushed_at}` is the ISO8601 format
+`YYYY-mm-ddThh:mm:ss.uuuZ` of the `pushed_at` datetime.
+
+For example, for the above `version` 1 `details` object,
+the JSON string to be signed would be `"LSPS5:Version 1:
+Reason 1 at time 2023-04-25T14:27:03.842Z."`
+
+For `version` 1, the following `reason` codes are defined:
+
+* `1` - The client has a pending incoming payment.
+* `2` - The LSP wants to recover some of the liquidity from
+  the client.
+* `3` - The LSP wants to change feerates for channels or
+  other management actions.
+* `4` - The LSP wants to end services to the client.
+
+The client application server, on receiving a `POST`
+request on its URI:
+
+* MUST check the `GET` parameters of the URI for any
+  authorization that the specified `lsp` node ID has
+  been authorized to make the push notification.
+* MUST check the `GET` parameters of the URI for which
+  instance of the client application the notification
+  is for.
+* for `version` 1, MUST validate that the `details.zbase`
+  signs the message specified, based on `details.reason`
+  and `details.pushed_at`.
+* for `version` 1, MUST check if the given `lsp` has
+  given a `pushed_at` in a previous `POST`, and if so,
+  MUST check that the current `pushed_at` is strictly
+  later than the latest previous `pushed_at`; if not, it
+  MUST ignore the push notification.
+* MAY filter out push notifications by other policies,
+  such as too many push notifications in a time frame,
+  or rejecting particular `reason` codes.
+
+> **Rationale** The `pushed_at` check is intended to
+> avoid replay attacks, where some third party is able
+> to copy the JSON object sent by the LSP and replays
+> it in order to frame the LSP for spamming the client.
+
+The `POST` response is unspecified.
+The LSP MUST ignore the response to the `POST`.
```

Comment:
Other webhook systems defined http code `200` as successfully received. Everything else is an error. 
We don't have a retry concept in this case but having a way to detect that the `client application server` received the message successfully is helpful when debugging. A http `500` would clearly indicate something is off.

Reply:
Maybe?

```Markdown
The LSP MUST consider an HTTP response code `200` as a success.
The LSP SHOULD log any non-`200` response code as an
unusual event, together with any response body and headers,
and MUST otherwise ignore the response.
```

Reply:
Perfect.

Reply:
Done.